app/battle.py

```diff
@@ -0,0 +1,49 @@
+from app.preparation.armour import Armor
+from app.preparation.weapon import Weapon
+from app.preparation.potion import Potion
+
+
+def prepare_knight(knight: dict) -> None:
+    Armor(knight).put_on_armour()
+    Weapon(knight).tool_up()
+    Potion(knight).use_potion()
+
+
+def battle(knights: dict) -> dict:
+    for knight in knights.values():
+        prepare_knight(knight)
+
+    # BATTLE:
+    # 1 Lancelot vs Mordred:
+    lancelot = knights["lancelot"]
+    mordred = knights["mordred"]
+
+    lancelot["hp"] -= mordred["power"] - lancelot["protection"]
+    mordred["hp"] -= lancelot["power"] - mordred["protection"]

```

Comment:
The calculation for reducing `hp` is correct, but it is repeated for both battles. Consider creating a helper function to handle this logic to adhere to the DRY (Don't Repeat Yourself) principle.